root.c

```diff
@@ -2223,6 +2224,14 @@ static w_root_t *root_resolve(const char *filename, bool auto_watch,
   }
 
   watch_path = w_realpath(filename);
+
+  w_lstat(filename, &st, true);
```

Comment:
it will make sense to make this block contingent on getting a valid result from `w_realpath` so that the realpath call can catch the real ENOENT and we can distinguish between that and a bad casing problem.

```
if (watch_path) {
   struct stat st;
   w_stat(..)
}
```